packages/abi-gen/src/index.ts

```diff
@@ -143,6 +143,7 @@ for (const abiFileName of abiFileNames) {
     const contextData = {
         contractName: namedContent.name,
         ctor,
+        ABI: JSON.stringify(ABI),
         methods: methodsData,
         events: eventAbis,
```

Comment:
@xianny how come we're able to pass in the `eventAbis` directly but not the ABI's?

Reply:
We pass in the `eventAbis` (and `methodAbis`) and then iterate through them to pull out certain named properties, e.g. `name`. We can do that too with the contract ABI. What we can't do is reference the whole object as we iterate. If we iterate through an array of primitives, we can print each primitive with `{{this}}`. But if we iterate through an array of objects and try to reference `{{this}}`, we'll get `Object object`. 

![Screen Shot 2019-06-28 at 11 20 37 AM](https://user-images.githubusercontent.com/8582774/60362847-c9b90200-9996-11e9-8e80-79036bbac207.png)
![Screen Shot 2019-06-28 at 11 21 24 AM](https://user-images.githubusercontent.com/8582774/60362880-e2291c80-9996-11e9-8580-7351e9df8a2c.png)

I haven't been able to find a single place that explicitly says you can't do this, but after combing through the docs and experimenting with different kinds of handlebars block helpers, I think it's going to be a waste of time to continue.

Reply:
Not sure what to do about the comment. I could make a stack overflow post with more details and link to it from the comment?

Reply:
The workaround would be to hardcode all the object properties. I guess that's fine - ABI typedef is pretty stable.

Reply:
Ok! Np, then, was just curious but it makes sense that we're using the entire ABI differently.